gunicorn/arbiter.py

```diff
@@ -473,6 +473,8 @@ def manage_workers(self):
         Maintain the number of workers by spawning or killing
         as required.
         """
+        orig_num_workers = self.num_workers
+
         if len(self.WORKERS.keys()) < self.num_workers:
```

Comment:
Yeah—I believe it can be removed synchronously as long as `worker.tmp.close()` is called. I wasn't sure if there was another reason why they might be left around for later reaping. (For example, reaping does not call `self.cfg.worker_exit(self, worker)` but trying to kill a dead worker does; that might also be a bug?)